### PR TITLE
Add exit code constants in promtool

### PR DESCRIPTION
**Pull request by JessicaGreben**
This PR adds constants for the exit codes in promtool for readability. 

It was not clear to me what exit code 3 was suppose to indicate in promtool main.go. Back in this PR, it is documented to be "one or more lint issues": https://github.com/prometheus/prometheus/pull/2605#discussion_r111264047

Signed-off-by: jessicagreben <jessicagrebens@gmail.com>

### CODE REVIEW

File: cmd/promtool/main.go [hunk 1]
```diff
@@ -56,6 +56,13 @@ import (
 	"github.com/prometheus/prometheus/scrape"
 )
 
+const (
+	successExitCode = 0
+	failureExitCode = 1
+	// exit code 3 is used for "one or more lint issues detected".
```
Comment (LeviHarrison) [inline cmd/promtool/main.go:62]: nit: capitalize comment

```suggestion
	// Exit code 3 is used for "one or more lint issues detected".
```

**Review by LeviHarrison — COMMENTED** [review]
Thanks!

The exit code constants also need to be added to [RulesUnitTest](https://github.com/prometheus/prometheus/blob/main/cmd/promtool/unittest.go#L42-L62) and [CheckSD](https://github.com/prometheus/prometheus/blob/main/cmd/promtool/sd.go#L40-L108), and since those two are public so should the constants.

At that point, we might want to consider making the constants a separate type over `int` (like `PromtoolExitCode`) for ease of use by external consumers. That may be too far though.

**Comment by JessicaGreben** [comment]
@LeviHarrison Thanks for catching the rulesUnitTest and checkSD file, I will add those.  I don't think it needs to be a type over int. I would rather leave as is if thats ok with you.

**Comment by LeviHarrison** [comment]
Sounds good

**Comment by JessicaGreben** [comment]
@darshanime Do you know what exit code 2 is suppose to mean for in the `promtool/sd.go` file ([code](https://github.com/prometheus/prometheus/blob/main/cmd/promtool/sd.go#L77))?

**Comment by LeviHarrison** [comment]
I think exit code 2 fits there because `cfg.NewDiscoverer()` is validating the discovery config.

Overall though, I'm not sure if it makes sense to have the exit code 2 anywhere because it's not uniform across Promtool commands. Either we use it (when appropriate) in all commands or none.

**Comment by JessicaGreben** [comment]
@LeviHarrison I agree. I will use exit 2 where appropriate but I will change the other uses of it to exit 1.

**Comment by LeviHarrison** [comment]
It seems that `configErrExitCode` is currently defined as when there's an error with `config.LoadFile()`, which is only executed in that place in `sd.go` and in `main.go`:

https://github.com/prometheus/prometheus/blob/59f7ef06d0002e818c73642c963d49f3ccf3d70e/cmd/promtool/main.go#L340-L343

In that instance though, an exit code isn't directly returned, so it ends up being the normal `failureExitCode`. In light of this and just general ambiguity around the code (why can't it just be `failureExitCode`?), I think we should just drop it altogether.

**Comment by JessicaGreben** [comment]
@LeviHarrison sounds good. I will make that change.

**Comment by LeviHarrison** [comment]
By the way, @roidelapluie will changing the exit codes for the `check service-discovery` command be a problem for our stability guarantees? It's made a release already. I think exit codes are a relatively minor change, but I thought I'd check with you anyway.

**Comment by roidelapluie** [comment]
In this case this is more a bugfix than a change.

**Review by LeviHarrison — APPROVED** [review]
LGTM, thanks!

**Pull request merged**